cognite/neat/_store/_base.py

```diff
@@ -292,6 +301,11 @@ def read(
         if not self.rules:
             warnings.warn("Rules not found in graph store!", stacklevel=2)
             return
+        if self.multi_type_instances:
```

Comment:
I realize that we have 3 read methods, causing some duplicated warnings and checks. Could this be pulled out in its own method? Something like `_validate_ready_for_read`